text/0000-crate-quarantine.md

```diff
@@ -0,0 +1,156 @@
+- Feature Name: `crate-quarantine`
+- Start Date: 2023-07-25
+- RFC PR: [rust-lang/rfcs#3464](https://github.com/rust-lang/rfcs/pull/3464)
+
+# Summary
+[summary]: #summary
+
+Add an off-by-default quarantine system to crates.io that can be used to prevent crate versions from being immediately published to the crate index based on rules defined by the crates.io team.
+
+# Motivation
+[motivation]: #motivation
+
+Packaging ecosystems aren't warm, friendly places any more. PyPI has repeatedly been forced to deal with attacks of spam[^pypi-2021-03][^pypi-2021-06] and malware[^pypi-2023-05]. In March 2023, an analysis indicated that more than half the packages uploaded to NPM were SEO spam[^npm-2023-03], and vendors such as jFrog have previously found significant numbers of malware packages on NPM[^npm-malware].
+
+So far, crates.io has been relatively fortunate to only have isolated incidents[^rustdecimal], but the increasing popularity of Rust only makes it more and more likely that we will also be the target of similar spam and malware attacks.
+
+There are mitigating factors at play: in particular, our reliance on GitHub for authentication, while problematic from purist FOSS and ecosystem diversity perspectives, means that we benefit from GitHub's machinery in terms of detecting bad actors.
+
+Regardless, we need to be able to prevent malicious packages from being uploaded, both in emergency situations (for example, mitigating spam attacks) and in more mundane cases where we can confidently identify a potentially malicious user. This RFC proposes an off-by-default quarantine approach that allows us to examine packages before they are published when it is considered warranted.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This project will add a new admin feature to crates.io that allows crate publishes matching specific properties to be placed into a quarantine state instead of being immediately published.
+
+A version in the quarantined state will have its `.crate` file published to S3 to simplify review, and will appear in crates.io API responses as an unpublished crate, but will _not_ be added to the index — and will therefore _not_ be installable via `cargo` — until it has been reviewed and approved.
+
+`cargo publish` will also be modified to report when an uploaded crate version has been placed into quarantine. 
+
+Once a crate version has been quarantined, then a member of the crates.io team will review the quarantined crate, ideally within 1 week[^sla]. Once reviewed, the crate may be either approved and published to the index, or rejected and deleted. Either way, the crate owner(s) will be notified by e-mail, assuming crates.io has a verified e-mail available for their account.
+
+Should the crate owner(s) disagree with the quarantine action, they may appeal to the moderation team.
```

Comment:
Do we want to specify how long we will wait for an answer of the crate owner before potentially deleting a quarantined crate?